### PR TITLE
improved test coverage of microstrip plugin

### LINKED ISSUE #1692: Improve test coverage of microwave models
Testing coverage is low for the microwave models:

> tidy3d/plugins/microwave/models/coupled_microstrip.py	49	42	0	14%
> tidy3d/plugins/microwave/models/microstrip.py	64	52	0	19%

### PR DESCRIPTION
**Pull request by dmarek-flex**
adding some unit tests and removing deprecated typing

the models part is at 100% now

### CODE REVIEW

File: tidy3d/plugins/microwave/models/coupled_microstrip.py [hunk 1]
```diff
@@ -51,7 +50,7 @@ def _z0_even_odd(
     e_eff_odd: float,
     z0: float,
     e_eff: float,
-) -> Tuple[float, float]:
+) -> tuple[float, float]:
```
Comment (weiliangjin2021) [inline tidy3d/plugins/microwave/models/coupled_microstrip.py:53]: Use `Tuple` to be consistent with the rest of code?

File: tests/test_plugins/test_microwave.py [hunk 2]
```diff
@@ -303,3 +308,58 @@ def impedance_of_stripline(width, height):
...
+    )
+    assert np.isclose(Z_even, 101.5, rtol=0.01).all()
+    assert np.isclose(Z_odd, 38.5, rtol=0.01).all()
+    assert np.isclose(eps_even, 3.26, rtol=0.01).all()
+    assert np.isclose(eps_odd, 2.71, rtol=0.01).all()
+
+    (Z_even, Z_odd, eps_even, eps_odd) = coupled_microstrip.compute_line_params(
+        eps_r, w2, height, g2
+    )
+    assert np.isclose(Z_even, 71, rtol=0.01).all()
```
Comment (weiliangjin2021) [inline tests/test_plugins/test_microwave.py]: how about `np.allclose`?

**Review by weiliangjin2021 — APPROVED** [review]

File: tidy3d/plugins/microwave/models/coupled_microstrip.py (hunk 1 above)
Comment (dmarek-flex) [reply tidy3d/plugins/microwave/models/coupled_microstrip.py:53]: I think we are expecting to deprecate using those aliases. [see here](https://docs.python.org/3/library/typing.html#deprecated-aliases)

Is that right @tylerflex ?

Comment (tylerflex) [reply tidy3d/plugins/microwave/models/coupled_microstrip.py:53]: yea we should start using the builtins like `tuple`

File: tests/test_plugins/test_microwave.py [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import pytest
```
Comment (tylerflex) [inline tests/test_plugins/test_microwave.py:3]: can you add a module level docstring and docstrings for the individual tests?

Also you may want to run `ruff tests/test_plugins/test_microwave.py` as we'll start `ruff`ing test files from 2.7.0

**Review by tylerflex — APPROVED** [review]

File: tests/test_plugins/test_microwave.py (hunk 3 above)
Comment (tylerflex) [reply tests/test_plugins/test_microwave.py:3]: careful about ruffing the whole test directory or you'll get a lot of changes / errors, many of which are already covered #1707

**Comment by tylerflex** [comment]
I think we can merge this in @dmarek-flex ?

**Comment by dmarek-flex** [comment]
> I think we can merge this in @dmarek-flex ?

I just need to add docstrings to the individual tests, then I will merge.

**Comment by tylerflex** [comment]
sounds good, just be sure to squash to 1 commit, thanks!

**Pull request merged**